R/runDPCoA.R

```diff
@@ -141,11 +139,27 @@ setMethod("calculateDPCoA", c("ANY","ANY"), .calculate_dpcoa)
 #' @rdname runDPCoA
 setMethod("calculateDPCoA", signature = c("TreeSummarizedExperiment","missing"),
     function(x, ..., assay_name = abund_values, abund_values = exprs_values, 
-             exprs_values = "counts", dimred = NULL, n_dimred = NULL)
+             exprs_values = "counts", tree_name = "phylo")
     {
         .require_package("ade4")
+        # Check assay_name
+        .check_assay_present(assay_name, x)
+        # Check tree_name
+        .check_rowTree_present(tree_name, x)
+        #
+        # Get tree
+        tree <- rowTree(x, tree_name)
+        # Select only those features that are in the rowTree
+        whichTree <- rowLinks(x)[ , "whichTree"] == tree_name
+        if( any(!whichTree) ){
+            warning("Not all rows were present in the rowTree specified by 'tree_name'.",
+                    "'x' is subsetted.", call. = FALSE)
+            # Subset the data
+            x <- x[ whichTree, ]
```

Comment:
It might be that the rowTree(x, "something") does not include all the rows (rowTree(x, "another") has those) --> the data is subsetted to match the tree